tests/CMakeLists.txt

```diff
@@ -117,6 +122,18 @@ function(ConfigureTest TEST_NAME)
 
 endfunction()
 
+# Create an object library for the logger so that we don't have to recompile it. This is not
+# necessary if logging is unsupported since then the header-only implementation is sufficient.
+if(RMM_SUPPORTS_LOGGING)
+  add_library(rmm_test_logger OBJECT)
+  target_link_libraries(rmm_test_logger PRIVATE rmm_logger_impl)
+endif()
+
+# TODO: Should we use CPM here, or since it's just tests should we assume spdlog is provided in the
+# environment? Even if we want to use CPM, we should probably simplify the logic to not need to
+# recompile spdlog with the same flags as rmm.
```

Comment:
I'm wondering whether we should be using `get_spdlog` anywhere in RAPIDS anymore after everything is updated to use rapids-logger. We could simply make it BYOspdlog so that using rapids-logger in building RAPIDS libraries will fetch spdlog for you, but all consumers of that (including our own tests) are responsible for providing the spdlog library in the environment. That isn't generally an onerous requirement because we can get it from conda. I'd like to discourage usage of rapids_cpm_spdlog as much as possible after this so that libraries don't accidentally put spdlog symbols into their primary public binaries.

Reply:
No longer relevant.